src/test/java/org/ohdsi/authenticator/service/GoogleIapAuthenticationTest.java

```diff
@@ -0,0 +1,68 @@
+package org.ohdsi.authenticator.service;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ohdsi.authenticator.exception.AuthenticationException;
+import org.ohdsi.authenticator.model.UserInfo;
+import org.ohdsi.authenticator.service.support.GoogleIapTestUtils;
+import org.pac4j.core.credentials.UsernamePasswordCredentials;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+@ActiveProfiles(profiles = { "test", "test-iap" })
+public class GoogleIapAuthenticationTest extends BaseTest {
```

Comment:
What about a separate successful test for UserInfo resolution?

Reply:
done